src/H5private.h

```diff
@@ -2251,6 +2302,23 @@ H5_DLL herr_t H5CX_pop(void);
                         FUNC_ENTER_API_INIT(err);                                                            \
                         {
 
+/*
+ * Use this macro for API functions that shouldn't perform _any_ initialization
+ *      of the library or an interface, or push themselves on the function
+ *      stack, or perform tracing, etc.  This macro _only_ sanity checks the
+ *	API name itself.  Examples are: H5TSmutex_acquire,
+ *
+ */
+#define FUNC_ENTER_API_NAMECHECK_ONLY                                                                        \
+    {                                                                                                        \
+        {                                                                                                    \
+            {                                                                                                \
+                {                                                                                            \
+                    {                                                                                        \
+                        {                                                                                    \
+                            FUNC_ENTER_COMMON_NOERR(H5_IS_API(FUNC));                                        \
+                            {
+
```

Comment:
That's not pretty :)

Reply:
> That's not pretty :)

Yeah, sometimes the code reformatted makes some ugly decisions. :-/

Reply:
Well I'm also wondering why you need so many "{" in the first place...

Reply:
> Well I'm also wondering why you need so many "{" in the first place...

Each of the FUNC_ENTER / EXIT pairs needs to have a different # of brackets, so that they only match correctly, not matching a differently named ENTER / EXIT.   (Not my favorite method for enforcing the matching, so I'm open to other suggestions)

Reply:
Yeah I'm not sure, I feel like this would need some deeper cut probably. I don't really have the bandwidth right now to look more into it.